### PR TITLE
Add docker build

### PR DESCRIPTION
**Pull request by hnico21**
Added Dockerfiles.
Added docker.compose.yml
Added instructions to Build image with docker-compose and start container.

### CODE REVIEW

**Comment by MODSetter** [comment]
Hi @hnico21 
Thank you very much for doing this as I have very limited knowledge with docker. I will test this today and merge it by EOD. You can mark this PR ready for review if you are done with commits.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  backend:
+    build:
+      context: ./backend
+    ports:
+      - "8000:8000"
+    env_file:
+      - ./backend/.env
+    depends_on:
+      - db
+    networks:
```
Comment (hnico21) [inline docker-compose.yml:26]: when backend is pointing to the hostmaschine with localhost or 127.0.0.1 you need to add `priviliged: true` else the container points at itself

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  # Backend Service (FastAPI)
+  backend:
+    build:
+      context: ./backend
+    ports:
+      - "8000:8000"
+    env_file:
+      - ./backend/.env
+    depends_on:
+      - db
```
Comment (hnico21) [inline docker-compose.yml:25]: when backend is pointing to the hostmaschine with localhost or 127.0.0.1 you need to add `privileged: true` else the container points at itself

Comment (MODSetter) [reply docker-compose.yml:25]: Hi @hnico21 

I added privileged: true and exposed 5432 port for Postgres.
I am getting this error trace for backend docker container. 
Don't know why its not been able to connect with Postgres:

UPDATE: Never mind I just had to change localhost to db in envs of backend

Comment (hnico21) [reply docker-compose.yml:25]: Thats what I thought. My .env looks like this:
POSTGRES_DATABASE_URL=postgresql+psycopg2://your_postgres_user:your_postgres_password@db:5432/surfsense_db

**Pull request merged**